docs/ADD_NEW_EXCHANGE.md

```diff
@@ -303,22 +300,36 @@ go run documentation.go
 
 This will generate a readme file for the exchange which can be found in the new exchange's folder
 
+### Code Consistency Guidelines
+
+1. Exchange API function parameters and structs should default to using unsigned integers (e.g., uint64) instead of int or int64. Using int can be problematic since we target different architectures, and its size varies between 32-bit and 64-bit systems (e.g., I still run Raspberry Pi on a 32-bit architecture). Explicitly using uint64 ensures consistency and prevents issues like negative values where they don’t make sense. Additionally, many common strconv functions (e.g., FormatUint) default to uint64, making it a more natural choice over uint32, despite the potential memory savings.
+
+2. Exchange parameters for start and end times should use time.Time instead of int64 UNIX timestamps. Even if an exchange requires a UNIX timestamp, we should convert it within the function to maintain consistency and readability.
+
+3. Time usage within Exchange API requests and responses should default to UTC. There is an existing PR for this, but further work is needed to enforce UTC across all exchanges before merging.
+
+4. TestMain in exchanges must avoid API calls to keep test footprint as minimal as possible unless there's a good case as to why it was done
+
+5. Excessive exchange API method params should use structs
+
+6. Test deduplication should be the default approach for exchanges, image provided as an example diff
+
 ### Create functions supported by the exchange:
 
 #### Requester functions:
```

Comment:
Think this can go into `rest.go`

Reply:
You mean the requests?

Reply:
Don't you think it makes the code guide-line to put it into single.

Reply:
Sorry; I don't understand this sentence.

I think that the code that belongs in rest.go should live in rest.tmpl, and we should be telling the dev to review the file, instead of "copy and paste something like this".
It means the skeleton is already there, and we can generate working go code straight away.